jax/_src/clusters/slurm_init.py

```diff
@@ -0,0 +1,45 @@
+import os
+from jax._src.clusters import Cluster
+
+_JOBID_PARAM = 'SLURM_JOB_ID'
+_NODE_LIST = 'SLURM_STEP_NODELIST'
+_PROCESS_COUNT = 'SLURM_NTASKS'
+_PROCESS_ID = 'SLURM_PROCID'
+_LOCAL_PROCESS_ID = 'SLURM_LOCALID'
+
+class SlurmCluster(Cluster):
+  @classmethod
+  def is_env_present(cls):
+    return _JOBID_PARAM in os.environ
+
+  @classmethod
+  def get_coordinator_address(cls):
+    # Pick port in ephemeral range [(65535 - 2^3 + 1), 65535]
```

Comment:
Good catch, I meant to have a bigger range like the last `2^12` ports of the ephemeral range but forgot to update it. Yes you cannot use `/proc/sys/net/ipv4/ip_local_port_range` since the logic to get the port needs to lead to the exact same port on all the processes to be able to connect to the open port of the 1st process. Using `/proc/sys/net/ipv4/ip_local_port_range` may give you different answer on different nodes.